docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+##### Disadvantages
+- Can be harder to compute and filter
+    - Large number of events to aggregate
+    - Some events may not be relevant to user experience (external monitoring HTTP request probes for example)
+- May be hard to alert on, in case of low frequency phenomenons (if the event occurs twice a day, single failed event might cause 50% failure ratio on short time ranges)
+- May lead to accounting repeated events of one user leading to multiplication of the impact (user refreshing the page to see if its operational).
+
+## SLI types
+These are the most common types of SLIs we encountered, of course you can think of any others that suit you but
+if possible try to stick to these common ones for consistency.
+They are based on the SRE workbook, and it makes it easier for others to understand what the SLI means.
+
+For more inspiration see https://sre.google/workbook/implementing-slos/#slis-for-different-types-of-services
+
+### Availability
+#### Time based
+Also, often referred to as an _uptime_, a ratio of the time when the service is up to the time it is down.
+You should be careful what actually means the service _is down_.
+Mostly suitable for some continuously running computations or in cases when the event based SLI is hard
+or even impossible to evaluate reliably for example SLO for relational databases(latency od success of the event depends on the user query).
+
+#### Event based
+Probably most common SLI used mainly for web services computed as a ratio of failed/successful requests served by the service.
+Should generally describe ratio of failed events to its total count. 
+The Simplest example could be expressed as a PromQL `http_request_duration_seconds_count{service="foo", status_code=~"5.."} / ignoring(status_code) http_request_duration_seconds_count{service="foo"}`.
+
+> REST API requests are quite easy to observe but dealing with for example GraphQL, XmlRPC, gRPC might be challenging
+> since the result might be encoded in the request body or in GraphQL case one HTTP request might contain several responses
+> thus it's hard to just binary say success or failure.
+
+> Carefully decide what is the failure of your service and what is the client's fault.
+> It always depends on the service but a lot of 4xx HTTP statuses might be disputable.
+
+### Latency
+Similar to the availability SLI type, but instead of a success or failure we measure a latency of the event.
+Sticking to the example of REST API we would observe how long it takes to serve the request.
+
+Latency SLI is observed in the form of percentiles such as "95% of events take longer than 1.2s".
+
+> Keep in mind that a failed event might take way less time than a successful one. So you should consider if the failed events should be counted into your latency SLI or not.
+> Also, for example HTTP status 499 means the client gave up the waiting. Be careful if this is observed by the latency SLI or by the availability type.
```

Comment:
I am not sure if `499` status code is good example. Its not sandardized - only nginx uses it https://httpstatus.in/499/

Reply:
Ok, good point, thanks. I'll exclude the status code and just describe the situation :grimacing: 

```suggestion
> Also be careful for example on user side timeouts. It might be observed in both, the latency and availability SLI type.
```